app/src/test/java/com/github/alfabravo2013/readyforexams/domain/login/LocalDataSourceTest.kt

```diff
@@ -0,0 +1,91 @@
+package com.github.alfabravo2013.readyforexams.domain.login
+
+import com.github.alfabravo2013.readyforexams.util.Result
+import org.junit.jupiter.api.Test
+
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.extension.RegisterExtension
+import org.koin.dsl.module
+import org.koin.test.KoinTest
+import org.koin.test.inject
+import org.koin.test.junit5.KoinTestExtension
+
+internal class LocalDataSourceTest : KoinTest {
+    private val localDataSource by inject<LocalDataSource>()
+
+    private val defaultEmail = "test@test.com"
+    private val defaultPassword = "123456789"
+
+    @JvmField
+    @RegisterExtension
+    val koinTestExtension = KoinTestExtension.create {
+        modules(
+            module {
+                single { LocalDataSource() }
+            }
+        )
+    }
+
+    @Test
+    @DisplayName("Given non-unique email When signup Then Result.Failure")
```

Comment:
The good part of JUnit5 is that it allows you to create inner classes for tests. I suggest to create an inner class for any method at test that has several scenarios. It will allow you to structure the code better. For example your inner class SignUp {} will have two situations.